### PR TITLE
feat(#814): implement reporting unit details

### LINKED ISSUE #814: [DEV TASK]: Implement Reporting Unit details/confirmation page
### Task Summary

Implement the Reporting Unit details/confirmation page shown after successful create.

### Task Details

- Create `src/pages/reportingUnit/ReportingUnitDetailsPage.tsx` to display created RU details.
- Ensure the page reads the RU identifier from route params and fetches details via `reportingUnit.service.getReportingUnit(id)`.
- Render the confirmation copy verbatim from Figma and include links/actions as required.
- Add unit tests and an E2E check to confirm the redirect and visible confirmation copy.

### Affected Environments

- [x] Development
- [x] Test
- [x] Production

### Additional Context

- Confirmation copy Figma: https://www.figma.com/design/MYE7heempT56a77UnBZTvS/Waste-V2.0---Create-a-RU?node-id=337-2088&m=dev

### PR DESCRIPTION
**Pull request by paulushcgcj**
# Description

> **Context:** The backend and legacy API layer for this feature was shipped in PR #839 (issue #836).
> This PR completes the work by delivering the frontend implementation and a minor backend enhancement (grade field on `ReportingUnitDetailsDto`).

> **Note:** If the diff appears larger than expected, run `git rebase main` on the branch first.
> The branch was forked before #839 and #843 merged, so those commits inflate the diff until a rebase is done.

Completes the Reporting Unit Details feature (#814). When a user clicks a reporting
unit in the search results table, they are routed to a dedicated details page
showing tombstone metadata and legacy-system data links.

**Frontend**

- new `ReportingUnitDetails` page uses a TanStack Router loader (`loader.ts`) with Zod-validated responses from `reportingunit.service.ts`.
- `ReportingUnitDetailsTombstone` renders key tombstone fields. `LegacyDataTag` renders tooltip-wrapped external links to the legacy system. `WasteSearchTable` gains a row action that navigates to the details page.
- `applyGuards` utility composes HOC route guards declaratively.
- Supporting changes: `RedirectLinkTag` updated to control search param inheritance, `env.ts` Zod schema adds the `reporting-unit-details-enabled` flag, auth helpers gain `isLoggedIn` state and cleaner login/logout mocks for tests, and ESLint is updated to warn on unused variables.

**Backend** 

- adds `grade` field to `ReportingUnitDetailsDto` and updates the mapping in `ReportingUnitService`. Additive and non-breaking.

Fixes #814

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

# How Has This Been Tested?

- [x] New unit tests
- [x] New integrated tests
- [x] New end-to-end tests
- [x] Updated existing tests

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Feature-Flag Controlled Changes (Required if applicable)

If this PR adds, changes, or removes behavior behind a feature flag, this section is required.

- [x] Canonical key follows `<domain>-<capability>-enabled` → **`reporting-unit-details-enabled`**
- [x] Frontend and backend use the same canonical key (`FeatureFlag.REPORTING_UNIT_DETAILS_ENABLED` in Java; `reporting-unit-details-enabled` in `env.ts` Zod schema)
- [x] Tests cover enabled, disabled, and undefined behavior

If any required checkbox above is unchecked, stop and resolve before requesting review.

## Further comments

The `applyGuards` utility (`frontend/src/routes/applyGuards.ts`) was introduced to avoid manually chaining `withProtected(withOfflineSupport(...))` HOC calls on each route component.
It reads the `RouteDescription.guards` config object and composes only the applicable HOCs in a single pass — making the route tree declarative and easier to test in isolation.

The WireMock stub fixtures under `frontend/stub/__files/reporting-units/` were shipped in PR #839. This PR adds one new stub (`details-with-grade.json`) to cover the grade-field edge case introduced by the backend change above.

---

Thanks for the PR!

Deployments, as required, will be available below:
Any successful deployments (not always required) will be available [here](https://nr-waste-plus-40-frontend.apps.silver.devops.gov.bc.ca/)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-waste-plus/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-waste-plus/actions/workflows/merge.yml)

### CODE REVIEW

**Review by mamartinezmejia — APPROVED** [review]

File: backend/src/main/java/ca/bc/gov/nrs/hrs/service/ReportingUnitService.java [hunk 1]
```diff
@@ -64,8 +70,9 @@ public ReportingUnitDetailsDto getReportingUnitDetails(Long reportingUnitId) {
             clientInformation.clientStatusCode().getDescription()
         ),
         legacyClient.sampling(),
-        legacyClient.district()
+        legacyClient.district(),
+        new CodeDescriptionDto(null, null)
```
[the remark this answers is not in the document]
Comment (paulushcgcj) [reply backend/src/main/java/ca/bc/gov/nrs/hrs/service/ReportingUnitService.java:74]: This will stay like this until a future feature that populates this field
